providers/aws/resources/aws_cloudtrail.go

```diff
@@ -0,0 +1,237 @@
+			}
+
+			for i := range trailsResp.TrailList {
+				trail := trailsResp.TrailList[i]
+
+				// only include trail if this region is the home region for the trail
+				// we do this to avoid getting duped results from multiregion trails
+				if regionVal != toString(trail.HomeRegion) {
+					continue
+				}
+				args := map[string]*llx.RawData{
+					"arn":                        llx.StringData(toString(trail.TrailARN)),
+					"name":                       llx.StringData(toString(trail.Name)),
+					"isMultiRegionTrail":         llx.BoolData(toBool(trail.IsMultiRegionTrail)),
+					"isOrganizationTrail":        llx.BoolData(toBool(trail.IsOrganizationTrail)),
+					"logFileValidationEnabled":   llx.BoolData(toBool(trail.LogFileValidationEnabled)),
+					"includeGlobalServiceEvents": llx.BoolData(toBool(trail.IncludeGlobalServiceEvents)),
+					"snsTopicARN":                llx.StringData(toString(trail.SnsTopicARN)),
+					"cloudWatchLogsRoleArn":      llx.StringData(toString(trail.CloudWatchLogsRoleArn)),
+					"region":                     llx.StringData(toString(trail.HomeRegion)),
+				}
+
+				// trail.S3BucketName
+				if trail.S3BucketName != nil {
+					mqlAwsS3Bucket, err := NewResource(a.MqlRuntime, "aws.s3.bucket",
+						map[string]*llx.RawData{"name": llx.StringData(toString(trail.S3BucketName))},
+					)
+					if err != nil {
+						return nil, err
+					}
+					args["s3bucket"] = llx.ResourceData(mqlAwsS3Bucket, mqlAwsS3Bucket.MqlName())
+				}
+
+				// add kms key if there is one
+				if trail.KmsKeyId != nil {
+					mqlKeyResource, err := NewResource(a.MqlRuntime, "aws.kms.key",
+						map[string]*llx.RawData{"arn": llx.StringData(toString(trail.KmsKeyId))},
+					)
+					// means the key does not exist or we have no access to it
+					// dont err out, just assign nil
+					if err != nil {
+						args["kmsKey"] = llx.NilData
+					} else {
+						mqlKey := mqlKeyResource.(*mqlAwsKmsKey)
+						args["kmsKey"] = llx.ResourceData(mqlKey, mqlKey.MqlName())
+					}
+				}
+				if trail.CloudWatchLogsLogGroupArn != nil {
+					mqlLoggroup, err := NewResource(a.MqlRuntime, "aws.cloudwatch.loggroup",
+						map[string]*llx.RawData{"arn": llx.StringData(toString(trail.CloudWatchLogsLogGroupArn))},
+					)
+					// means the log group does not exist or we have no access to it
+					// dont err out, just assign nil
+					if err != nil {
+						args["logGroup"] = llx.NilData
+					} else {
+						mqlLog := mqlLoggroup.(*mqlAwsCloudwatchLoggroup)
+						args["logGroup"] = llx.ResourceData(mqlLog, mqlLog.MqlName())
+					}
+				}
+
+				mqlAwsCloudtrailTrail, err := a.MqlRuntime.CreateResource(a.MqlRuntime, "aws.cloudtrail.trail", args)
+				if err != nil {
+					return nil, err
+				}
+
+				res = append(res, mqlAwsCloudtrailTrail)
+			}
+			return jobpool.JobResult(res), nil
+		}
+		tasks = append(tasks, jobpool.NewJob(f))
+	}
+	return tasks
+}
+
+func (a *mqlAwsCloudtrailTrail) s3bucket() (*mqlAwsS3Bucket, error) {
```

Comment:
I do not understand why we need this. I read this as the AWS object just does not have a S3 bucket. Couldn't we just remove the field from the MQL resource? The same for the next two funcs.

Reply:
hmm, ill look